packages/koa-error-handler/index.js

```diff
@@ -1,13 +1,35 @@
-module.exports = () => async (ctx, next) => {
+const defaultRenderHtml = () =>
+  `<!DOCTYPE html>
+    <html prefix="og: http://ogp.me/ns#">
+      <head><title>Oops</title></head>
+      <body>
+        <h3>Oops something wrong happened...</h3>
+        <b>That's our fault, your did nothing wrong.</b>
+        <p>Please bear with us, our team is already working on it.</p>
+      </body>
+    </html>`;
+
+module.exports = ({ renderHtml } = {}) => async (ctx, next) => {
   try {
     await next();
   } catch (err) {
+    const contentType = ctx.accepts('html', 'json');
     ctx.status = 500;
-    ctx.body = {
-      statusCode: 500,
-      error: 'Internal Server Error',
-      message: 'An internal server error occurred'
-    };
+    if (typeof renderHtml === 'function' && contentType === 'html') {
+      ctx.set('Content-Type', 'text/html; charset=utf-8');
+      try {
+        ctx.body = await renderHtml(ctx);
+      } catch (renderError) {
+        console.error(renderError);
+        ctx.body = defaultRenderHtml();
```

Comment:
no need for a function here, defaultPlainHtml will do the job, maybe we could ask for { renderHtml and plainHtml, if renderHtml not defined we use plainHtml and if not defined too we use defaultPlainHtml ?

Reply:
Good point, but in these cases there is no risk of a breaking change?
For an api, it's not safe to have an html rendering.